src/api-services/user.service.js

```diff
@@ -20,16 +20,17 @@ export default {
     return Axios.post(RESOURCE_NAME.concat('/login'), userData)
   },
   confirmEmail (id, token) {
-    return Axios.patch(RESOURCE_NAME.concat(`/emailVerification/${id}`), { token }, {
-      headers: {
-        accept: 'application/json'
-      }
-    })
+    return Axios.patch(RESOURCE_NAME.concat(`/emailVerification/${id}`), token)
+  },
+  changePassword (id, userData) {
+    return Axios.patch(RESOURCE_NAME.concat(`/passwordRecovery/${id}`), userData)
+  },
+  generatePasswordResetToken (email) {
+    return Axios.patch(RESOURCE_NAME.concat(`/recoveryTokenGeneration`), email)
```

Comment:
Czemu nie?
/verify
/recovery
/token

Czy użytkownik musi widzieć cały brzydki url? Skoro kliknął na password recovery to raczej wie o co chodzi.

Reply:
Użytkownik tego nie będzie widział, chyba że wejdzie w dev toolsy i ruch sieci. To są requesty ajaxowe i one wywołują metody na moich kontrolerach API (np kontroler user i metoda email verification) a potem zwracają response. To co użytkownik będzie widział w URL jest w routerze.
![obraz](https://user-images.githubusercontent.com/15947565/66698138-97c87300-ecdb-11e9-8efc-d6dbb13125cb.png)

Reply:
Jak pozmieniasz linki w routerze, to jak użytkownik wyśle odpowiednie zapytanie za pomocą przycisku???? Nie rozumiem Cię w tym momencie. 
Jak sobie kliknę zresetuj hasło to mnie przekieruje na ekran RESET PASSWORD gdzie jest url z id i z tokenem i tak dalej.

Reply:
Reset hasła działa tak 
1. User klika przycisk Zapomniales hasło?  i zostaje przekierowany na route /forgotPassword
2. User w formularzu wpisuje email na który było założone konto. Klika wyślij, odpalana jest akcja generatePasswordToken. Na backendzie generowany jest token resetu hasła przypisywany jest użytkownikowi o emailu podanym w formularzu i zapisywany do bazy. Po wygenerowaniu tokenu zostaje wysłany mail na mail podany w formularzu. W wiadomości zawiera się link w postaci "adresFrontu/resetPassword/Id usera dla którego zostal wygenerowany token/Token który został wygenerowany" 
3.User wchodzi na maila klika link przedstawiony mu jest formularz w którym musi wpisać hasło i potwierdzić je. 
4. User klika przycisk zresetuj hasło. Z Urla pobierany jest token i id uzytkownika, na froncie wywoluje się akcję changePassword która przesyla na backend token i id pobrane z urla i hasło które sobie uzytkownik wybral.
5. Na backendzie sprawdza się czy istnieje user o podanym id, jesli istnieje to sprawdza sie czy token przeslany w requescie jest taki sam jak zapisany w bazie danych. jesli tak to ustawiane jest nowe haslo

Reply:
Więc jedynym efektem zmiany routa bedzie zmiana wyglądu URL'a w mailu i w pasku adresu przeglądarki